src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -251,14 +253,10 @@ def get_pyenv_root(env: Environment) -> str | None:
 async def python_bootstrap(
     python_bootstrap_subsystem: PythonBootstrapSubsystem, env_tgt: EnvironmentTarget
 ) -> PythonBootstrap:
-    if env_tgt.val is not None:
-        interpreter_search_paths = env_tgt.val[PythonInterpreterSearchPathsField].value
-        interpreter_names = env_tgt.val[PythonBootstrapBinaryNamesField].value
-        for opt in ("search_path", "names"):
-            python_bootstrap_subsystem.error_if_environment_mechanism_ambiguity(opt)
-    else:
-        interpreter_search_paths = python_bootstrap_subsystem.search_path
-        interpreter_names = python_bootstrap_subsystem.names
+
+    # TODO: use subsystems directly again once do overrides at subsystem contruct time.
+    interpreter_search_paths = get_option("search_path", python_bootstrap_subsystem, env_tgt)
+    interpreter_names = get_option("names", python_bootstrap_subsystem, env_tgt)
```

Comment:
We can still definitely land this PR without that change. And I'd probably encourage we do to make this more incremental and easier to review

Reply:
Yup, I think we should land with the `get_option` API and move on to making it more automatic in a follow-up.